### ISSUE TITLE
Feature request: add tf.layers.Group to group multiple layers under one name

### ISSUE THREAD

**Issue by bodokaiser** [tensorflow/tensorflow#13338]
Example usage (relevant for networks with skip connection i.e. u-net):

```python
encoder1 = tf.layers.Group([
  tf.layers.Conv2d(...),
  tf.layers.BatchNorm(),
  ActivationLayer(),
], name='encoder1')

encoder2 = tf.layers.Group([
  tf.layers.Conv2d(...),
  tf.layers.BatchNorm(),
  ActivationLayer(),
], name='encoder1')

inputs = outputs = tf.layers.Input(tensor=x)

for enc in [encoder1, encoder2]:
  outputs = enc(outputs)

encoders = tf.layers.Network(inputs, outputs)

# same for decoders
for i, dec in enumerate(decoders):
  outputs = dec(tf.concat([encoders.get_layer(f'encoder{i}').output, outputs], 3))
```

Currently you could either
* use `network.get_output_at()` however this requires to track node indices or parse `network.layers` if encoders are not symmetric (i.e. some encoder have no batch norm / dropout)
* inherit from `tf.layers.Layer` however documentation is not clear on how to "forward" variables for example

**Comment by bignamehyp** [comment]
In your example, you named your encoder2 'encoder1'

**Comment by drpngx** [comment]
@martinwicke WDYT?

**Comment by martinwicke** [comment]
Sorry, I don't understand what this does -- is this a version of Sequential?

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and this issue has an assignee.Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by yaroslavvb** [comment]
Closing since the interested parties seem to have moved on
